bot/exts/moderation/slowmode.py

```diff
@@ -44,16 +44,21 @@ async def get_slowmode(self, ctx: Context, channel: Optional[TextChannel]) -> No
         await ctx.send(f'The slowmode delay for {channel.mention} is {humanized_delay}.')
 
     @slowmode_group.command(name='set', aliases=['s'])
-    async def set_slowmode(self, ctx: Context, channel: Optional[TextChannel], delay: DurationDelta) -> None:
+    async def set_slowmode(self, ctx: Context, channel: Optional[TextChannel], delay: str) -> None:
         """Set the slowmode delay for a text channel."""
         # Use the channel this command was invoked in if one was not given
         if channel is None:
             channel = ctx.channel
 
         # Convert `dateutil.relativedelta.relativedelta` to `datetime.timedelta`
         # Must do this to get the delta in a particular unit of time
-        slowmode_delay = time.relativedelta_to_timedelta(delay).total_seconds()
+        try:
+            delay = await DurationDelta().convert(ctx, delay)
+
+        except BadArgument:
+            delay = relativedelta(seconds=0)
```

Comment:
This isn't necessarily the expected behavior. This behavior means that any invalid duration will reset the slowmode, while the behavior outlined in the issue is to just make an exception for "0s".

You could check for the value of the string, or you could make use of the converters and make the delay a union of `DuraitonDelta` and a `Literal["0s"]`